### PR TITLE
Corpus idf

### PR DESCRIPTION
**Pull request by gauravbaruah**
as part of sourcing-IDF-from-index and e2e experiments.

### CODE REVIEW

File: idf_baseline/experimental_settings.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+            cmd = '{} {}/{}.qrel run.{}.idfsim'.format(self.eval_cmd_root,
+                                                       self.qa_data, split, split)
+            out, err = self._run_cmd(cmd)
+            print(split, '------' )
+            # trec_eval scores
+            metrics = []
+            scores = []
+            for line in str(out).split('\n'):
+                if not line.strip().split(): continue
+                fields = line.strip().split()
```
Comment (rosequ) [inline idf_baseline/experimental_settings.py]: Put `continue` on the next line

File: idf_baseline/experimental_settings.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+                metrics.append(fields[0])
+                scores.append(fields[-1])
+            
+            # rbp_eval scores
+            cmd = '{} {}/{}.qrel run.{}.idfsim'.format(self.rbp_cmd_root,
+                                                       self.qa_data, split, split)
+            out, err = self._run_cmd(cmd)
+            for line in str(out).split('\n'):
+                if not line.startswith('p= 0.50'): continue
+                metrics.append('rbp_p0.5')
```
Comment (rosequ) [inline idf_baseline/experimental_settings.py]: continue on the next line

File: idf_baseline/experimental_settings.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+if __name__ == "__main__":
+    ap = argparse.ArgumentParser(description="Lists exerimental settings and runs experiments")
+    ap.add_argument("--list", help="lists all available experimental settings combinations",
+                    action="store_true")
+    ap.add_argument("--run", help='runs experimenal setting combination NUMBER(s)', 
+                    nargs="+", type=int)
+    ap.add_argument("--runall", help="runs all experiments in order", action="store_true")
+    ap.add_argument("indexPath", help="required for some combination of experiments")
+    ap.add_argument('qa_data', help="path to the QA dataset",
```
Comment (rosequ) [inline idf_baseline/experimental_settings.py]: change this to `index_path`?

File: idf_baseline/experimental_settings.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+# cmd="python qa-data-only-idf.py ../../data/TrecQA run --index ../../data/indices/index.qadata.pos.docvectors.keepstopwords/ --dash-split"
+# runeval "$cmd"
+
+# # ('idf_source:corpus-index', 'stop_stem:no', 'punctuation:remove', 'dash_words:keep')
+# cmd="python qa-data-only-idf.py ../../data/TrecQA run --index ../../data/indices/index.qadata.pos.docvectors.keepstopwords/ --stop-punct"
+# runeval "$cmd"
+
+# # ('idf_source:corpus-index', 'stop_stem:no', 'punctuation:remove', 'dash_words:split')
+# cmd="python qa-data-only-idf.py ../../data/TrecQA run --index ../../data/indices/index.qadata.pos.docvectors.keepstopwords/ --stop-punct --dash-split"
+# runeval "$cmd"
```
Comment (rosequ) [inline idf_baseline/experimental_settings.py]: delete the commented code

File: idf_baseline/qa-data-only-idf.py [hunk 5]
```diff
@@ -1,40 +1,96 @@
...
+import re
 import numpy as np
 from collections import defaultdict
 import string
+import subprocess
+import shlex
 
 import nltk
-nltk.download('stopwords')
+nltk.download('stopwords', quiet=True)
```
Comment (rosequ) [inline idf_baseline/qa-data-only-idf.py]: So, this doesn't throw any warnings if not connect to the Internet, is it?

File: sm_cnn/bridge.py [hunk 6]
```diff
@@ -86,9 +116,16 @@ def rerank_candidate_answers(self, question, answers, idf_json):
...
+        for term in question.split():
+            if term not in term_idfs:
+                term_idfs[term] = 0.
+
         for answer in answers:
             answer = self.parse(answer)
-
+            for term in answer.split():
+                if term not in term_idfs:
+                    term_idfs[term] = 0.
```
Comment (rosequ) [inline sm_cnn/bridge.py]: Why is there  a `.` after 0 in line 121 and 127?

File: idf_baseline/experimental_settings.py (hunk 4 above)
Comment (gauravbaruah) [inline idf_baseline/experimental_settings.py]: Sorry about that :P

File: idf_baseline/qa-data-only-idf.py (hunk 5 above)
Comment (gauravbaruah) [inline idf_baseline/qa-data-only-idf.py]: Still gives a download error if not connected to the internet. However, the program proceeds correctly of the stopwords exist locally (have been pre-downloaded or pre-installed).

File: sm_cnn/bridge.py (hunk 6 above)
Comment (gauravbaruah) [inline sm_cnn/bridge.py]: To signify that its a float.

**Pull request merged**